python/python/res/job_queue/job_manager.py

```diff
@@ -221,6 +227,12 @@ def _loadJson(self, json_file_name):
         self._ensureCompatibleJobList()
         self._buildJobMap()
 
+        for job in self.job_list:
+            self.job_status.add_job( ForwardModelJobStatus(job.get("name")))
+
+        # "Monkey-patching" the job object by attaching a status object.
+        for i in range(len(self.job_list)):
+            self.job_list[i]["status"] = self.job_status[i]
```

Comment:
I feel this monkey patching is a bit unnecessary. The `job_list` here seems relatively static, and it seems that every entry in `self.job_list` is in a one-to-one correspondence with entries of `self.job_status`. So, the status of the `i`-th job in `self.job_list` can be always retrieved with `self.job_status[i]`

Reply:
OK - that might be; I'll look into it.